R/MsExperiment.R

```diff
@@ -202,3 +202,60 @@ setMethod("readMsObject", signature(object = "MsExperiment",
           function(object, param, ...) {
               readAlabasterMsExperiment(path = param@path, ...)
           })
+
+
+################################################################################
+##
+## MetaboLights readMsObject
+##
+################################################################################
+#' @rdname MetaboLightsParam
+#' @importFrom utils menu
+setMethod("readMsObject",
+          signature(object = "MsExperiment",
+                    param = "MetaboLightsParam"),
+          function(object, param, ...) {
+              url <- file.path(
+                  "https://ftp.ebi.ac.uk/pub/databases/metabolights/studies/public/",
+                  param@studyId, "/")
+              ## Retrieve the HTML content using curl system command
+              html_content <- system(paste("curl -s", url), intern = TRUE) ##idk if that works for all OS
```

Comment:
Interesting solution you have here - I implemented it differently in *MsBackendMetaboLights*. I'll check what would be safer to use. Also, since we both need the same functionality to list assays for a MetaboLights entry maybe we should have a dedicated helper function for this. I would suggest to maybe put that into the *MsBackendMetaboLights* or *MetabCoreUtils* package.

Reply:
Ok ye, that part i'm really unsure, i found this way robust and quite efficient, I would love if you explain later what solution is better and why :)